src/new_hope/diagonal_difference.clj

```diff
@@ -0,0 +1,50 @@
+(ns new-hope.diagonal-difference
+  (:require [midje.sweet :refer :all]
+            [criterium.core :as crit]))
+
+;https://www.hackerrank.com/challenges/diagonal-difference
+
+(def matrix [[11 2 4] [4 5 6] [10 8 -12]])
+
+(defn element [m i j] (nth (nth m i) j))
+
+(defn first-sum [m, n]
+  (loop [sum 0
+         line 0]
+    (if (= n line)
+      sum
+      (recur (+ sum (element m line line)) (inc line)))))
+
+(defn second-sum [m n]
+  (loop [sum 0
+         line 0]
+    (if (= n line)
+      sum
+      (recur (+ sum (element m line (- n (+ line 1)))) (inc line)))))
+
+(defn abs-of-diff [x y] (Math/abs (- x y)))
```

Comment:
I don't see any sense in this, almost no typing saved and no additional abstraction introduced.